crates/ibc/src/hosts/tendermint/upgrade_proposal/context.rs

```diff
@@ -0,0 +1,57 @@
+//! Helper Context for handling upgrade client proposals.
+//!
+//! Currently. this interface has been defined to support Tendermint-based
+//! chains. You can check out a sample implementation in the
+//! [Basecoin-rs](https://github.com/informalsystems/basecoin-rs) repository.
+//! If it proves to be generic enough, we may move it to the ICS02 section.
+
+use alloc::boxed::Box;
+
+use super::Plan;
+use crate::core::ics02_client::client_state::ClientState;
+use crate::core::ics02_client::consensus_state::ConsensusState;
+use crate::core::ics02_client::error::UpgradeClientError;
+use crate::core::ics24_host::path::UpgradeClientPath;
+
+/// Helper context for validating upgrades, providing methods to retrieve
+/// upgrade plans and upgraded client and consensus states.
+pub trait UpgradeValidationContext {
```

Comment:
I now think it makes more sense to bring `upgrade_client_proposal_handler()` (now defined in basecoin) back in ibc-rs. Otherwise, we have these traits defined, and nothing that uses them. And I now understand that this handler/traits are optional; chains are always free to implement upgrades however they want. So we can start with the `upgrade_client_proposal_handler` we currently have, and iteratively improve it over time

Reply:
This was chain-dependent at some point while implementing, but with the latest code, right, it can now be placed in IBC-rs.
[2950ab7](https://github.com/cosmos/ibc-rs/pull/680/commits/2950ab7232f7d0514204334fa772faf6c4a43da4)